un_sdg/match_variables.py

```diff
@@ -0,0 +1,113 @@
+"""Creates a `variable_replacements.json` file of {old variable id} to
+{new variable id} key-value pairs.
+`variable_replacements.json` is used in `suggest_chart_revisions.py` for
+determining which charts to update.
+Usage:
+    python -m un_sdg.match_variables
+"""
+
+import os
+import simplejson as json
+import logging
+from typing import List
+import pandas as pd
+
+from db import get_connection
+from db_utils import DBUtils
+from un_sdg import OUTPATH
+
+DEBUG = True
+DEBUG_HOST = 'http://localhost:3030'
+UPDATE_DB = False
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG if DEBUG else logging.INFO)
+
+def main():
+    connection = get_connection()
+    with connection.cursor() as cursor:
+        db = DBUtils(cursor)
+        # retrieves old and new datasets
+        df_old_datasets = get_datasets(db=db, new=False)
+        df_new_datasets = get_datasets(db=db, new=True)
+
+        # retrieves old and new variables
+        df_old_vars = get_variables(db=db, dataset_ids=df_old_datasets['id'].tolist())
+        df_new_vars = get_variables(db=db, dataset_ids=df_new_datasets['id'].tolist())
+        df_vars = pd.merge(df_old_vars, df_new_vars, on='name', how='inner',
+                            suffixes=['_old', '_new'], validate='1:1')
+        assert df_vars.id_old.notnull().all() and df_vars.id_new.notnull().all()
+        old_var_id2new_var_id = df_vars.dropna(subset=['id_old', 'id_new']) \
+                                    .set_index('id_old')['id_new'] \
+                                    .squeeze() \
+                                    .to_dict()
+        if not os.path.exists(OUTPATH):
+            os.makedirs(OUTPATH)
+        with open(os.path.join(OUTPATH, 'variable_replacements.json'), 'w') as f:
+            json.dump(old_var_id2new_var_id, f)
```

Comment:
indent=2 or so might be nice here so that the json file is not a single line and diffs nicer if the file is overwritten in a later git commit